libraries/AP_Scripting/generator/src/main.c

```diff
@@ -273,12 +274,29 @@ void handle_header(void) {
     error(ERROR_HEADER, "Header %s was already included on line %d", name, node->line);
   }
 
+  char * key_word = next_token();
+  char *depends = NULL;
+  if (key_word != NULL) {
+    if (strcmp(key_word, keyword_depends) != 0) {
+      error(ERROR_HEADER, "Headers only support depends keyword, got %s", key_word);
+    }
+    depends = strtok(NULL, "");
+    if (depends == NULL) {
+      error(ERROR_HEADER, "Expected a depends string for %s", name);
+    }
```

Comment:
```suggestion
    if (strcmp(key_word, keyword_depends) == 0) {
        depends = strtok(NULL, "");
        if (depends == NULL) {
          error(ERROR_HEADER, "Expected a depends string for %s", name);
        }
    } else {
      error(ERROR_HEADER, "Recieved an unsupported keyword on a header: %s", key_word);
    }
```

We should be using this pattern instead of handling everything first, rather then checking for the one case you currently care about. The problem is when we come back to add something else to this the code churn will be much larger. We should strive to keep a general pattern of handling things first, then falling to the error state last.

Reply:
made suggested change